Library/ViewModels/BackerDashboardViewModel.swift

```diff
@@ -0,0 +1,239 @@
+
+  /// Emits a ProfileProjectsType to configure the Saved Projects Controller.
+  var configureSavedProjectsController: Signal<ProfileProjectsType, NoError> { get }
+
+  /// Emits a CGFloat to set the top constraint of the embedded views when the sort bar is hidden or not.
+  var embeddedViewTopConstraintConstant: Signal<CGFloat, NoError> { get }
+
+  /// Emits when to present Messages.
+  var goToMessages: Signal<(), NoError> { get }
+
+  /// Emits the project and ref tag when should go to project page.
+  var goToProject: Signal<(Project, [Project], RefTag), NoError > { get }
+
+  /// Emits when to navigate to Settings.
+  var goToSettings: Signal<(), NoError> { get }
+
+  /// Emits an index to pin the indicator view to a particular button view with or without animation.
+  var pinSelectedIndicatorToPage: Signal<(Int, Bool), NoError> { get }
+
+  /// Emits an attributed string for the saved button title label.
+  var savedButtonTitleText: Signal<String, NoError> { get }
+
+  /// Emits a boolean whether saved projects container should be hidden or not.
+  var savedProjectsAreHidden: Signal<Bool, NoError> { get }
+
+  /// Emits an index of the selected button to update all button selected states.
+  var setSelectedButton: Signal<Int, NoError> { get }
+
+  /// Emits when should scroll to the project item or row position.
+  var scrollToProject: Signal<Int, NoError> { get }
+
+  /// Emits a boolean whether the sort bar is hidden or not.
+  var sortBarIsHidden: Signal<Bool, NoError> { get }
+}
+
+public protocol BackerDashboardViewModelType {
+  var inputs: BackerDashboardViewModelInputs { get }
+  var outputs: BackerDashboardViewModelOutputs { get }
+}
+
+public final class BackerDashboardViewModel: BackerDashboardViewModelType, BackerDashboardViewModelInputs,
+  BackerDashboardViewModelOutputs {
+
+  // swiftlint:disable:next function_body_length
+  public init() {
+    let user = Signal.merge(
+      self.viewDidLoadProperty.signal,
+      self.viewWillAppearProperty.signal.filter(isFalse).skip(first: 1).ignoreValues()
+      )
+      .switchMap { _ in
+        AppEnvironment.current.apiService.fetchUserSelf()
+          .prefix(SignalProducer([AppEnvironment.current.currentUser].compact()))
+          .demoteErrors()
+    }
+
+    self.avatarURL = user.map { URL(string: $0.avatar.large ?? $0.avatar.medium) }
+
+    self.backedButtonTitleText = user
+      .map { user in
+        localizedString(
+          key: "projects_count_backed",
+          defaultValue: "%{projects_count} backed",
+          count: user.stats.backedProjectsCount ?? 0,
+          substitutions: ["projects_count": Format.wholeNumber(user.stats.backedProjectsCount ?? 0)]
+        )
+        .replacingOccurrences(of: " ", with: "\n")
+    }
+
+    self.backerLocationText = user.map { $0.location?.displayableName ?? "" }
+
+    self.backerNameText = user.map { $0.name }
+
+    self.configureBackedProjectsController = self.viewDidLoadProperty.signal
+      .map { .backed }
```

Comment:
since this output doesnt have any logic i think it's fine to `configureWith` right when we find the embedded controller:

```swift
self.backedProjectsViewController = self.childViewControllers
  .flatMap { $0 as? ProfileProjectsViewController }
  .first
self.backedProjectsViewController.configureWith(projectsType: .backed)
```

etc...